olp-cpp-sdk-core/include/olp/core/network2/NetworkRequest.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <olp/core/CoreApi.h>
+
+namespace olp {
+namespace network2 {
+
+/**
+ * @brief The NetworkRequest class represents network request abstraction for a
+ * HTTP request
+ */
+class CORE_API NetworkRequest final {
+ public:
+  using RequestBodyType = std::shared_ptr<const std::vector<std::uint8_t>>;
+  using RequestHeadersType = std::vector<std::pair<std::string, std::string>>;
+
+  enum class HttpVerb {
+    GET = 0,
+    POST = 1,
+    HEAD = 2,
+    PUT = 3,
+    DEL = 4,
+    PATCH = 5,
+  };
+
+  /**
+   * @brief NetworkRequest constructor
+   * @param[in] url - HTTP request URL
+   */
+  explicit NetworkRequest(std::string url);
+
+  /**
+   * @brief Get all HTTP headers
+   * @return vector of HTTP headers
+   */
+  const RequestHeadersType& GetHeaders() const;
+  /**
+   * @brief Add extra HTTP header
+   * @param[in] name - header name
+   * @param[in] value - header value
+   * @return reference to *this
+   */
+  NetworkRequest& WithHeader(std::string name, std::string value);
+
+  /**
+   * @brief Get request URL
+   * @return request URL
+   */
+  const std::string& GetUrl() const;
+  /**
+   * @brief Set request URL
+   * @param[in] url - request URL
+   * @return reference to *this
+   */
+  NetworkRequest& WithUrl(std::string url);
+
+  /**
+   * @brief Get HTTP method
+   * @return HTTP method
+   */
+  HttpVerb GetVerb() const;
+  /**
+   * @brief Set HTTP method
+   * @param[in] verb - HTTP method
+   * @return reference to *this
+   */
+  NetworkRequest& WithVerb(HttpVerb verb);
+
+  /**
+   * @brief Get request body
+   * @return shared pointer to vector that contains request body
+   */
+  RequestBodyType GetBody() const;
+  /**
+   * @brief Set request body
+   * @param[in] body - shared pointer to vector that contains request body
+   * @return reference to *this
+   */
+  NetworkRequest& WithBody(RequestBodyType body);
+
+ private:
+  HttpVerb verb_ = HttpVerb::GET;  /// HTTP request method
```

Comment:
I would prefer to use {} for in-class initialization as it is also used everywhere else.